SU-HSDO/suhumsci
# SHS-5560:  Skip to Main Content

# READY FOR REVIEW

## Summary
- Add "Main Content Anchor Target" block provided by SWS
- Add custom template and theming for "Main Content Anchor Target"
- Add update hooks to add block to all existing default and overridden layouts
- Remove `main-content` id from all node pages
- Update page main content container to use `<main>` element
- Add fallback js to add the `main-content` id if not present in page

## Need Review By (Date)


## Urgency
medium

## Steps to Test
1. Visit default layout builder configuration for all content types. Confirm that the "Main Content Anchor Target" is present in all cases
2. Find a node with an overridden layout configuration (*). Click on the "Layout" tab and confirm that the "Main Content Anchor Target" is present
3. Select some random pages and tab to the "Skip to Main Content" link. Click it or press enter to confirm that it works as expected
4. Remove the "Main Content Anchor Target" from a default or overridden layout. Visit the page and confirm that the js fallback works by using the "Skip to Main Content" link
5. Inspect any page and confirm that there's a `<main>` element wrapping the page content (everything except the header and footer)

(*) You can find the ids of the nodes with overridden layout executing the following query in the SQL console:

```
SELECT entity_id FROM node__layout_builder__layout;
```

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Humsci Basic PR Checklist](https://github.com/SU-HSDO/suhumsci/blob/develop/docs/HumsciBasicPRChecklist.md)
